### PR TITLE
Use flutter.js bootstrapping

### PR DESCRIPTION
**Pull request by johnpryan**
Attempt 2 at landing https://github.com/dart-lang/dart-pad/pull/2792

### CODE REVIEW

File: pkgs/dartpad_ui/firebase.json [hunk 1]
```diff
@@ -83,6 +83,15 @@
             }
           ]
         },
+        {
+          "source": "/canvaskit/chromium/canvaskit.wasm",
```
Comment (devoncarew) [inline pkgs/dartpad_ui/firebase.json:87]: This firebase CORS config looks reasonable to me. I wonder - do we expect to need to allowlist additional resources in the future? For this filename to change?

I also don't know if we're holding flutter web uniquely, or if this resource + CORS pattern should be part of general 'hosting a flutter web app' guidance.

**Review by devoncarew — APPROVED** [review]

File: pkgs/dartpad_ui/firebase.json (hunk 1 above)
Comment (johnpryan) [reply pkgs/dartpad_ui/firebase.json:87]: I think we will need to keep evaluating how we're configuring CORS and reevaluate how to simplify our configuration once we loosen things up enough to get them working.

We don't have good docs on how to configure CORS for Flutter web + Wasm, but I'm still not sure if this is a Flutter web specific configuration or if this is due to how we're sandboxing the iframe element.

cc @eyebrowsoffire @ditman @kevmoo

**Comment by johnpryan** [comment]
Just to close the loop, this fixed the CORS error.

There's still some optimization we could do here, we are still throwing away a lot of the assets that we could be sharing between compilations, but this gets us migrated to the new bootstrapping system.

Thanks @ditman for the help!

File: pkgs/dartpad_ui/firebase.json (hunk 1 above)
Comment (ditman) [reply pkgs/dartpad_ui/firebase.json:87]: With the incoming `--wasm` mode, there's a whole host of "CORS" cousins that will need this type of configuration + documentation, like [COEP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy) or [CORP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy). This file is cool (but of course it depends on the hosting/CDN that the developer picks)!

**Pull request merged**